crates/router/src/core/payments.rs

```diff
@@ -3690,60 +3698,202 @@ pub async fn decide_multiplex_connector_for_normal_or_recurring_payment<F: Clone
+                network_tokenization_supported_connectors,
+            );
 
-            let mut connector_choice = None;
+            let action_type = decide_action_type(
+                state,
+                key_store,
+                is_connector_agnostic_mit_enabled,
+                is_network_tokenization_enabled,
+                &payment_method_info,
+                &filtered_nt_supported_connectors,
+            )
+            .await;
 
-            for connector_data in connectors {
-                let merchant_connector_id = connector_data
-                    .merchant_connector_id
-                    .as_ref()
-                    .ok_or(errors::ApiErrorResponse::InternalServerError)
-                    .attach_printable("Failed to find the merchant connector id")?;
+            match action_type {
+                Some(ActionType::NetworkTokenWithNetworkTransactionId(token_exp_data)) => {
+                    logger::info!(
+                        "using network_tokenization with network_transaction_id for MIT flow"
+                    );
 
-                if is_network_transaction_id_flow(
-                    state,
-                    is_connector_agnostic_mit_enabled,
-                    connector_data.connector_name,
-                    payment_method_info,
-                ) {
-                    logger::info!("using network_transaction_id for MIT flow");
                     let network_transaction_id = payment_method_info
                         .network_transaction_id
                         .as_ref()
-                        .ok_or(errors::ApiErrorResponse::InternalServerError)?;
+                        .ok_or(errors::ApiErrorResponse::InternalServerError)
+                        .attach_printable("Failed to featch the network transaction id")?;
 
                     let mandate_reference_id =
-                        Some(payments_api::MandateReferenceId::NetworkMandateId(
-                            network_transaction_id.to_string(),
+                        Some(payments_api::MandateReferenceId::NetworkTokenWithNTI(
+                            payments_api::NetworkTokenWithNTIRef {
+                                network_transaction_id: network_transaction_id.to_string(),
+                                token_exp_month: token_exp_data.token_exp_month,
+                                token_exp_year: token_exp_data.token_exp_year,
+                            },
                         ));
+                    let connector_data = filtered_nt_supported_connectors
+                        .first()
+                        .ok_or(errors::ApiErrorResponse::IncorrectPaymentMethodConfiguration)
+                        .attach_printable(
+                            "no eligible connector found for token-based MIT payment",
+                        )?;
+
+                    let connector_choice =
+                        Some((connector_data.clone(), mandate_reference_id.clone()));
```

Comment:
We are creating Some here and again in the line using get_required_value function take some out which is not required